Source/Core/RequestScheduler.js

```diff
@@ -0,0 +1,256 @@
+/*global define*/
+define([
+        './Check',
+        './defined',
+        './defineProperties',
+        './Heap',
+        './isBlobUri',
+        './isDataUri',
+        './RequestState',
+        '../ThirdParty/when'
+    ], function(
+        Check,
+        defined,
+        defineProperties,
+        Heap,
+        isBlobUri,
+        isDataUri,
+        RequestState,
+        when) {
+    'use strict';
+
+    /**
+     * Tracks the number of active requests and prioritizes incoming requests.
+     *
+     * @exports RequestScheduler
+     *
+     * @private
+     */
+    function RequestScheduler() {
+    }
+
+    /**
+     * The maximum number of simultaneous active requests. Un-throttled requests do not observe this limit.
+     * @type {Number}
+     * @default 50
+     */
+    RequestScheduler.maximumRequests = 50;
+
+    /**
+     * The maximum size of the priority heap. This limits the number of requests that are sorted by priority. Only applies to requests that are not yet active.
+     * @type {Number}
+     * @default 20
+     */
+    RequestScheduler.priorityHeapSize = 20;
+
+    /**
+     * Specifies if the request scheduler should throttle incoming requests, or let the browser queue requests under its control.
+     * @type {Boolean}
+     * @default true
+     */
+    RequestScheduler.throttle = false;
+
+    /**
+     * When true, log statistics to the console every frame
+     * @type {Boolean}
+     * @default false
+     */
+    RequestScheduler.debugShowStatistics = false;
+
+    function sortRequests(a, b) {
+        // Sort requests by higher screen space error and then closer distance
+        if (a.screenSpaceError !== b.screenSpaceError) {
```

Comment:
Also, not scoped for this PR, but I wanted to experiment with effectively increasing the SSE when the camera is moving quickly (e.g., for camera flights) so tiles are not requested and then out of view when they are returned.  Perhaps the request scheduler will be the right place to add this logic since it would work system-wide and still allow the tileset/terrain/imagery to render with high SSE if it is already loaded.